slsim/Sources/source.py

```diff
@@ -20,26 +31,75 @@ def __init__(self, source_dict, variability_model=None, kwargs_variability=None)
          words, code search for quantities in source_dict with these names and creates
          a dictionary and this dict should be passed to the Variability class.
         :type kwargs_variability: list of str
+        :param kwargs_peak_mag: range of the peak magnitude
+        :type kwargs_peak_mag: dict. eg: kwargs_peak_mag={"peak_mag_min": m_min,
+         "peak_mag_max": m_max}
+        :param lightcurve_time: time period for lightcurve.
+        :type lightcurve_time: astropy unit object. egs: 10*u.day, 10*u.year.
         """
         self.source_dict = source_dict
         if kwargs_variability is not None:
             kwargs_variab_extracted = {}
-            for element in kwargs_variability:
-                if element in self.source_dict.colnames:
-                    if (
-                        isinstance(self.source_dict[element], np.ndarray)
-                        and self.source_dict[element].ndim == 2
-                        and self.source_dict[element].shape[0] == 1
-                    ):
-                        kwargs_variab_extracted[element] = self.source_dict[
-                            element
-                        ].reshape(-1)
-                    else:
-                        kwargs_variab_extracted[element] = self.source_dict[element]
+            kwargs_variability_list = [
```

Comment:
can you in general in the lines of code below write more what is happening. You can also put placeholder TODOs in there as well